internal/kubeclient/scheme.go

```diff
@@ -0,0 +1,61 @@
+package kubeclient
+
+import (
+	argocd "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/pkg/errors"
+	authzv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubescheme "k8s.io/client-go/kubernetes/scheme"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+)
+
+var kargoSchemes = runtime.SchemeBuilder{
+	corev1.AddToScheme,
+	kargoapi.AddToScheme,
+}
+
+func newScheme(extraSchemes ...func(*runtime.Scheme) error) (*runtime.Scheme, error) {
+	scheme := runtime.NewScheme()
+	for _, addToScheme := range extraSchemes {
+		if err := addToScheme(scheme); err != nil {
+			return nil, errors.Wrap(err, "add api to scheme")
+		}
+	}
+	return scheme, nil
+}
+func NewAPIScheme() (*runtime.Scheme, error) {
+	return newScheme(kargoSchemes...)
+}
+
+func NewAppManagerScheme() (*runtime.Scheme, error) {
+	return newScheme(
+		corev1.AddToScheme,
+		argocd.AddToScheme,
+	)
+}
+
+func NewCLIScheme() (*runtime.Scheme, error) {
+	return newScheme(kargoSchemes...)
+}
+
+func NewKargoScheme() (*runtime.Scheme, error) {
+	return newScheme(kargoSchemes...)
+}
+
+func NewKargoManagerScheme() (*runtime.Scheme, error) {
+	return newScheme(kargoSchemes...)
+}
+
+func NewKubernetesScheme() (*runtime.Scheme, error) {
+	return newScheme(kubescheme.AddToScheme, kargoapi.AddToScheme)
+}
+
+func NewGarbageCollectorScheme() (*runtime.Scheme, error) {
+	return newScheme(kargoSchemes...)
+}
```

Comment:
This function is used by two packages:

* `cmd/controlplane`
* `internal/garbage`

A tiny amount of refactoring would make this used in only by `internal/garbage`.

Specifically, the garbage collector's "constructor" could supply the client (and the scheme it uses) itself instead of relying on `cmd/controlplane` to inject it.

After that, this could be DRYed up using a package-private helper function.

I think this also points to a smell that's recurring for most components... we probably don't need `cmd/controlplane` _ever_ injecting clients or schemes into internal components that could reasonably build those themselves, which might be tidier.